### DISCUSSION TITLE
Multi drawer and do not move drawer when new one opens

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#51092] [Q&A] [answered]
Hello, I am trying to have a drawer that opens from another drawer, but without moving the parent drawer, how is this possible with AntD?

**Comment by petercat-assistant** [comment]
To achieve a multi-drawer setup in Ant Design where opening a new drawer does not move the parent drawer, you can use the `Drawer` component with some custom styling and positioning. Here is an example of how you can accomplish this:

```jsx
import React, { useState } from 'react';
import { Drawer, Button } from 'antd';

const MultiDrawerExample = () => {
  const [visibleParent, setVisibleParent] = useState(false);
  const [visibleChild, setVisibleChild] = useState(false);

  const showParentDrawer = () => {
    setVisibleParent(true);
  };

  const closeParentDrawer = () => {
    setVisibleParent(false);
  };

  const showChildDrawer = () => {
    setVisibleChild(true);
  };

  const closeChildDrawer = () => {
    setVisibleChild(false);
  };

  return (
    <div>
      <Button type="primary" onClick={showParentDrawer}>
        Open Parent Drawer
      </Button>
      <Drawer
        title="Parent Drawer"
        placement="right"
        closable={false}
        onClose={closeParentDrawer}
        visible={visibleParent}
        width={400}
      >
        <Button type="primary" onClick={showChildDrawer}>
          Open Child Drawer
        </Button>
        <Drawer
          title="Child Drawer"
          placement="right"
          closable={false}
          onClose={closeChildDrawer}
          visible={visibleChild}
          width={300}
          style={{ position: 'absolute', right: 0, top: 0 }}
        >
          <p>Content of the child drawer</p>
        </Drawer>
      </Drawer>
    </div>
  );
};

export default MultiDrawerExample;
```

In this example:
- The parent drawer is opened by clicking the "Open Parent Drawer" button.
- Inside the parent drawer, there is another button to open the child drawer.
- The child drawer is positioned absolutely within the parent drawer to prevent it from moving the parent drawer when it opens.

This setup ensures that the parent drawer remains in place while the child drawer opens.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by kiner-tang** [comment, marked as answer]
Maybe you can try to set `push=false` in parent drawer to prevent the parent drawer moving.
docs: https://ant.design/components/drawer#api
<img width="1467" alt="image" src="https://github.com/user-attachments/assets/b4102d57-dd7b-478f-93cd-c0a9b7e443ff">

https://stackblitz.com/edit/react-urp3ua?file=demo.tsx

**Reply by bombillazo** [reply]
Wow, I totally missed that prop! Thank you!
